### ISSUE TITLE
Add support for iOS 16.4 beta webkit safari web push notifications

### ISSUE THREAD

**Issue by hmkas** [humhub-contrib/fcm-push#17]
With the introduction of the long-awaited support for web push notifications in safari for iOS in the 16.4 beta 1 update will be able to work just as good on iPhones as on android devices.

However webkit has some requirements for it to work:

- Web Push on iOS and iPadOS uses the same Apple Push Notification service that powers native push on all Apple devices. You do not need to be a member of the Apple Developer Program to use it. Just be sure to allow URLs from *.push.apple.com if you are in control of your server push endpoints.

**Comment by luke-** [comment]
Thanks for the information, that is really very interesting and nice that we can soon offer push for IOS devices.

**Comment by Vander-Ng** [comment]
@luke- Hi Luke, any update on this?

**Comment by zrknlzr** [comment]
I tried looking into this as i need this feature for a bigger project with multiple Humhub instances too.

But it seems that the notification dialog is triggered by Humhub itself and this plugin has not much say in this. AFAIK this file is responsible for opening the browser API dialog on non-iOS platforms https://github.com/humhub/humhub/blob/b58125e818579b7549bbc0da67ea4dbf703eea19/static/js/desktop-notify-min.js

Webpush on iOS **only** works in a Progressive Web App ("Add to Homescreen") and needs to be triggered by the user within the web interface **not** the normal browser API dialog. Currently the browser API dialog just fires as soon as someone opens up the login screen so iOS just denies it by default i think.

@luke- The iOS version of the [app](https://github.com/humhub/app) is coming at some point i assume so this feature most likely won't get much priority?

**Comment by luke-** [comment]
@zrknlzr The IOS app should already have a working push notification support.

---

But it would be nice to have a button in the notification settings to ask for the permission for push notifications again.

**Comment by Vander-Ng** [comment]
> @zrknlzr The IOS app should already have a working push notification support.
> 
> But it would be nice to have a button in the notification settings to ask for the permission for push notifications again.

Thanks Luke.  Have you tried it?  Is it working just like an iOS app notification?  Can you share an article in Humhub community on how to enable it and how well it works?  I think many members would like to know this, thank you.

**Comment by luke-** [comment]
Yes, the push notifications work with the https://github.com/humhub/app app and the FCM push module. No special configuration is required for IOS.

**Comment by jannis6023** [comment]
Hi together, is there anything new in this topic here? We would love to use humhub, but push notifications are a must-have!
